core/src/main/java/com/yahoo/imapnio/async/data/ExtendedModifiedSinceTerm.java

```diff
@@ -0,0 +1,92 @@
+package com.yahoo.imapnio.async.data;
+
+import com.sun.mail.imap.IMAPMessage;
+
+import javax.annotation.Nullable;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.search.SearchTerm;
+
+/**
+ * Find messages that have been modified since a given MODSEQ value
+ * with optional entry name and entry type.
+ * Relies on the server implementing the CONDSTORE extension
+ * (<A HREF="http://www.ietf.org/rfc/rfc7162.txt">RFC 7162</A>).
+ */
+public final class ExtendedModifiedSinceTerm extends SearchTerm {
+
+    /** The given modification sequence. */
+    private final long modSeq;
+
+    /** The name of the metadata item. */
+    private final String entryName;
+
+    /** The type of metadata item. */
+    private final String entryType;
+
+    /**
+     * Constructor.
+     *
+     * @param modSeq modification sequence number
+     */
+    public ExtendedModifiedSinceTerm(final long modSeq) {
+        this(modSeq, null, null);
```

Comment:
Do not use nulls and avoid nullable annotations for parameters of constructors/methods... Instead annotate the fields of the class as nonnull/nullable. do this everywhere